django/thunderstore/api/cyberstorm/tests/test_package.py

```diff
@@ -0,0 +1,123 @@
+        count = 0
+        while count < n:
+            yield PackageVersionFactory.build(
+                name=package.name,
+                package=package,
+                is_active=True,
+                version_number=f"{count}.0.0",
+            )
+            count += 1
+
+    def package_gen(n, teams):
+        count = 0
+        while count < n:
+            team = random.choice(teams)
+            namespace = team.get_namespace()
+            p = PackageFactory.build(
+                is_active=True,
+                is_deprecated=False,
+                owner=team,
+                namespace=namespace,
+                name=f"{team.name}_package_{count}",
+            )
+            yield p
+            count += 1
+
+    def package_listing_gen(packages, communities):
+        count = 0
+        combos = list(itertools.product(range(len(packages)), range(len(communities))))
+        random.shuffle(combos)
+        while count < community_multiplier or count > len(combos):
+            pc, cc = combos[count]
+            # TODO: Add categories. Categories need to be added in community gen and grabbed here.
+            # TODO: Add review status. Review requirement needs to be added in community gen and all status types here.
+            # TODO: Add parametrized NSFW.
+            p_l = PackageListing(
+                community=communities[cc],
+                package=packages[pc],
+            )
+            yield p_l
+            count += 1
+
+    # Populate test DB
+    coms = Community.objects.bulk_create(
+        [
+            Community(name=f"Test_{com_count}", identifier=f"test_{com_count}")
+            for com_count in range(1, (community_multiplier + 1))
+        ]
+    )
+    teams = Team.objects.bulk_create(
+        [
+            Team(name=f"Test_Team_{t_count}")
+            for t_count in range(1, (team_multiplier + 1))
+        ]
+    )
+    pkgs = Package.objects.bulk_create(package_gen(package_multiplier, teams))
+    p_versions = []
+    for p in pkgs:
+        p_versions += package_version_gen(package_version_multiplier, p)
+    PackageVersion.objects.bulk_create(p_versions)
+    if len(pkgs) > 2:
+        pkgs[2].versions.first().dependencies.set(
+            [pkgs[0].versions.first(), pkgs[1].versions.first()]
+        )
+        pkgs[2].versions.first().save()
+    pkg_listings = PackageListing.objects.bulk_create(package_listing_gen(pkgs, coms))
+
+    for pl in pkg_listings:
+        # Update latest versions
+        pl.package.handle_updated_version(None)
+        response = client.get(
+            f"/api/cyberstorm/c/{pl.community.identifier}/p/{pl.package.namespace.name}/{pl.package.name}/",
+            HTTP_HOST=community_site.site.domain,
+        )
+        assert response.status_code == 200
+        # TODO: Add actual testing of the data
+        # Maybe run it through deserializer?
+
+
+@pytest.mark.django_db
+def test_PackageAPIView__get__can_be_viewed_by_user__False() -> None:
```

Comment:
Continuing my earlier comment regarding the naming, while there's no one correct way to do this, something like `test_PackageAPIView__rejected_listing__is_not_shown` would be more readable and descriptive to me.

Reply:
The project already has a somewhat established convention for test naming FWIW, see my comment here: https://github.com/thunderstore-io/Thunderstore/pull/846/files#r1238891332